app/routes.py

```diff
@@ -98,3 +104,31 @@ def handle_exception(e):
 
     # now you're handling non-HTTP exceptions only
     return render_template("500_generic.html", e=e), 500
+
+#caching all the dataone original sources in memory (only update the datasources after 120 secs)
+@cache.cached(timeout=120, key_prefix='cache_original sources for dataone')
```

Comment:
does this do it *every* 120 seconds, or 120 seconds after startup?

Reply:
from the documentation it says it does it every 120 seconds, i can increase the time out interval if that would be okay